### PR TITLE
Product management

### PR DESCRIPTION
**Pull request by 5t4mp5**
-create product
-edit product from main screen
-click edit button to see single product
-add/remove categories from single product page
--add categories to system from main product page

### CODE REVIEW

File: server/routes/product.js [hunk 1]
```diff
@@ -98,14 +98,16 @@ router.delete('/:id', (req, res, next) => {
...
+    // return Promise.all([ Product.create(req.body),
+    //                      Category.findOne({where: {name: req.body.categoryName}}) ])
+    // .then(([product, category]) => {
+    //     product.update({categoryId: category.id});
+    //     return product;
+    // })
+    req.body.stockStatus = req.body.quantity ? 'in stock' : 'out of stock'
+    Product.create(req.body)
+        .then((product) => res.json(product))
+        .catch((error) => console.log(error));
```
Comment (chem2016) [inline server/routes/product.js:110]: Should we call catch(next)?

File: src/SingleProductForm.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            id: null,
+            errors: []
+        };
+    }
+
+    load = () => {
+        if(this.props.products){
+            const product = this.props.products.find(_product => _product.id === this.props.match.params.productId);
+            if(product){
+                this.setState({
```
Comment (chem2016) [inline src/SingleProductForm.js:28]: Should be done by this.setState({...product, errors: []})

File: src/SingleProductForm.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                    quantity: product.quantity,
+                    description: product.description,
+                    imgUrl: product.imgUrl,
+                    productNumber: product.productNumber,
+                    id: product.id,
+                    errors:[]
+                });
+                getProductCats(product.id)
+                        .then(categories => this.setState({ categories }))
+                        .catch(e => e.response.data.errors);
```
Comment (chem2016) [inline src/SingleProductForm.js:40]: should we do catch(e=> this.setState({errors: e.response.data.errors}))

**Review by ami-domadia — APPROVED** [review]

**Review by chem2016 — COMMENTED** [review]
I have several comments

**Pull request merged**